API/controlsClass/controlsClass.m

```diff
@@ -286,5 +357,313 @@
+                end
+            end
+
+            % Sets the values the for NS parameters
+            obj.Nlive = inputBlock.Nlive;
+            obj.Nmcmc = inputBlock.Nmcmc;
+            obj.propScale = inputBlock.propScale;
+            obj.nsTolerance = inputBlock.nsTolerance;
+            obj.parallel = inputBlock.parallel;
+            obj.calcSldDuringFit = inputBlock.calcSldDuringFit;
+            obj.resamPars = inputBlock.resamPars;
+            obj.display = inputBlock.display;
+        end
+
+        function obj = processDreamInput(obj, varargin)
+            % Parses Dream keyword/value pairs and sets the properties of the class.
+            %
+            % obj.processDreamInput('param', 'value')
+            %
+            % The parameters that can be set when using Dream procedure are
+            % 1) nSamples
+            % 2) nChains
+            % 3) lambda
+            % 4) pUnitGamma
+            % 5) boundHandling
+            % 6) parallel
+            % 7) calcSldDuringFit
+            % 8) resamPars
+            % 9) display
+
+            % The default values for Dream
+            defaultNSamples = 50000;
+            defaultNChains = 10;
+            defaultLambda = 0.5;
+            defaultPUnitGamma = 0.2;
+            defaultBoundHandling = boundHandlingOptions.Fold.value;
+            defaultParallel = parallelOptions.Single.value;
+            defaultCalcSldDuringFit = false;
+            defaultResamPars = [0.9 50];
+            defaultDisplay = displayOptions.Iter.value;
+
+            % Creates the input parser for the Dream parameters
+            p = inputParser;
+            addParameter(p,'nSamples',  defaultNSamples,   @isnumeric);
+            addParameter(p,'nChains',   defaultNChains,    @isnumeric);
+            addParameter(p,'lambda', defaultLambda,  @isnumeric);
+            addParameter(p,'pUnitGamma',   defaultPUnitGamma,    @isnumeric);
+            addParameter(p,'boundHandling',   defaultBoundHandling, @(x) isText(x) || isenum(x));
+            addParameter(p,'parallel',  defaultParallel,   @(x) isText(x) || isenum(x));
+            addParameter(p,'calcSldDuringFit',   defaultCalcSldDuringFit,    @islogical);
+            addParameter(p,'resamPars', defaultResamPars,  @isnumeric);
+            addParameter(p,'display',   defaultDisplay,    @(x) isText(x) || isenum(x));
+            properties = varargin{:};
+
+            % Parses the input or raises invalidParameter error 
+            try
+                parse(p, properties{:});
+                inputBlock = p.Results;
+            catch ME
+                if (strcmp(ME.identifier,'MATLAB:InputParser:UnmatchedParameter'))
+                    throw(invalidParameter('Only nSamples, nChains, lambda, pUnitGamma, boundHandling, parallel, calcSldDuringFit, resamPars and display can be set while using the DREAM procedure'));
+                else
+                    rethrow(ME)
+                end
+            end
```

Comment:
I will change the exception part.

I agree there is a lot of repetition (Hence the missed typos in the docs after copy and pasting).  I left the functions like this because parser are defined in a similar way in the other classes. I could not think of a way to cleanly abstract out the addParameter() and the set value step into a more general form that I could then reuse for each of the different procedures. I think it can be done by passing list to functions. But I thought it might become less readable.

Reply:
Yes, I agree with that. Apart from the try-catch block, there was nothing obvious to me that would clearly improve the code whilst also reducing repetition. I also agree it could be done in ways which make the code less readable, which I'd also be inclined to avoid.